File: Sentence Smash.cpp
#include <iostream>
#include <vector>
#include <string>

std::string smashWords(const std::vector<std::string>& words) {
    std::string sentence;
    for (const std::string& word : words) {
        sentence += word + " ";
    }
    if (!sentence.empty()) {
        sentence.pop_back();
    }
    return sentence;
}
int main() {
    std::vector<std::string> words;
    std::string input;
    while (true) {
        std::cout << "Enter a word (or press Enter to finish): ";
        std::getline(std::cin, input);
        if (input.empty()) {
            break;
        }
        words.push_back(input);
    }
    std::string result = smashWords(words);

    // Print the result
    std::cout << "Resulting sentence: " << result << std::endl;
    return 0;
}

File: sentenceSmash.js
function smashWords(words) {
    // Join the words with spaces and return the sentence
    return words.join(' ');
}

// Usage with dynamic input from the user
let inputWords = [];
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

function getInput() {
    readline.question('Enter a word (or type "done" to finish): ', word => {
        if (word.toLowerCase() === 'done') {
            const sentence = smashWords(inputWords);
            console.log('Resulting sentence:', sentence);
            readline.close();
        } else {
            // Add the word to the inputWords array and continue to get more words
            inputWords.push(word);
            getInput();
        }
    });
}

// Start getting input from the user
getInput();
